### ISSUE TITLE
[Bug]: notification not supported

### ISSUE THREAD

**Issue by Antiarchitect** [mattermost/desktop#2935]
### Checks before filing an issue

- [X] This issue doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/mattermost-server/issues).
- [X] I have checked the [issue tracker](https://github.com/mattermost/desktop/issues) and have not found an issue that matches the one I'm filing.
- [X] This issue is not a troubleshooting question. Troubleshooting questions go here: https://forum.mattermost.com/c/trouble-shoot/16.
- [X] This issue is not a feature request. You can request features and make product suggestions here: https://mattermost.com/suggestions/.
- [X] This issue reproduces on the most recent [stable version](https://github.com/mattermost/desktop/releases/latest), or the most recent [prerelease version](https://github.com/mattermost/desktop/releases) of the Mattermost Desktop App.
- [X] I have read the [contribution guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

### Mattermost Desktop Version

5.5.0

### Operating System

NixOS 23.11

### Mattermost Server Version

8.1.7

### Steps to reproduce

Some time ago desktop notifications just stopped working. It could be an MMDesktop update or anything.

### Expected behavior

Desktop notifications work as expected

### Observed behavior

Desktop notifications not work and I see error messages in logs

### Log Output

```shell
[2023-12-20 13:44:13.099] [error] [Notifications] notification not supported
[2023-12-20 13:45:45.560] [error] [Notifications] notification not supported
[2023-12-20 13:48:29.668] [error] [Notifications] notification not supported
[2023-12-20 13:52:48.231] [error] [Notifications] notification not supported
[2023-12-20 13:53:52.359] [error] [Notifications] notification not supported
```

### Additional Information

NixOS + swaywm (wayland)
Chrome + web client: notifications work as expected

**Comment by devinbinnie** [comment]
@Antiarchitect This message is returned by Electron by called the [isSupported](https://www.electronjs.org/docs/latest/api/notification#notificationissupported) function on the `Notification` object. So it's likely that's returning false due to something not supported by your system, or potentially an issue with Electron.

Given that it used to work, it's possible support was dropped or a bug was introduced. If you can pinpoint which MM Desktop App version this used to work on, we can find the Electron version and submit a ticket.

**Comment by Antiarchitect** [comment]
Hmm... I see - will try to investigate if it's possible to install older package versions in NixOS somehow.

**Comment by devinbinnie** [comment]
Closing issue for now, since this issue lies with Electron and not the Desktop App.

**Comment by Antiarchitect** [comment]
Just tried to do Hello World with Electron Fiddle. And Electron 26 and 28 notifications work just fine:

```
// Modules to control application life and create native browser window
const { app, BrowserWindow, Notification } = require('electron')
const path = require('path')

function createWindow () {
  // Create the browser window.
  const mainWindow = new BrowserWindow({
    width: 800,
    height: 600,
    webPreferences: {
      preload: path.join(__dirname, 'preload.js')
    }
  })

  // and load the index.html of the app.
  console.log("Notifications: " + Notification.isSupported())
  new Notification({
  title: 'Hello',
  body: 'World'
  }).show()
  mainWindow.loadFile('index.html')
  // Open the DevTools.
  // mainWindow.webContents.openDevTools()
}

// This method will be called when Electron has finished
// initialization and is ready to create browser windows.
// Some APIs can only be used after this event occurs.
app.whenReady().then(() => {
  createWindow()

  app.on('activate', function () {
    // On macOS it's common to re-create a window in the app when the
    // dock icon is clicked and there are no other windows open.
    if (BrowserWindow.getAllWindows().length === 0) createWindow()
  })
})

// Quit when all windows are closed, except on macOS. There, it's common
// for applications and their menu bar to stay active until the user quits
// explicitly with Cmd + Q.
app.on('window-all-closed', function () {
  if (process.platform !== 'darwin') app.quit()
})

// In this file you can include the rest of your app's specific main process
// code. You can also put them in separate files and require them here.
```

And console shows:
```
Electron v26.6.8 started.
Notifications: true
```

```
Electron v28.2.1 started.
Notifications: true
```

![image](https://github.com/mattermost/desktop/assets/50209/31087612-0ee6-4ec7-82e1-d6f7c0f9971e)

It seems like it's not an Electron problem at all.

**Comment by devinbinnie** [comment]
@Antiarchitect Does this work with Desktop App v5.6? This might have been fixed in a patch release to Electron (v5.5.0 is on Electron v26.1.0, v5.6.0 is on v27.0.2)

**Comment by Antiarchitect** [comment]
Will test these exact Electron versions with Fiddle tomorrow

**Comment by Antiarchitect** [comment]
@devinbinnie Both (v26.1.0 and v27.0.2) works just fine. Could it be MM sends more sophisticated notification than just simple text title + text body?

**Comment by devinbinnie** [comment]
I wouldn't think so. The call we make that generates the log you're seeing looks like this:
```
import {Notification} from 'electron';

if (!Notification.isSupported()) {
    log.error('notification not supported');
    return;
}
```

The only thing I can think of at this point is that there's something in the compiled version, or some permission that your OS is denying to the compiled version that is causing this. Since you were able to make it work in a Fiddle, would you be able to try running it in development mode and see if it reproduces there? If so, would you be open to looking into a fix? If not, then it's likely something with the OS blocking it.

**Comment by devinbinnie** [comment]
Looking at that it looks like `npm i` failed somewhere since it's saying the Electron binary doesn't exist.

**Comment by Antiarchitect** [comment]
`npm i` output seems viable:

```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: webpack-filter-warnings-plugin@1.2.1
npm WARN Found: webpack@5.71.0
npm WARN node_modules/webpack
npm WARN   dev webpack@"5.71.0" from the root project
npm WARN   21 more (@storybook/react-docgen-typescript-plugin, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer webpack@"^2.0.0 || ^3.0.0 || ^4.0.0" from webpack-filter-warnings-plugin@1.2.1
npm WARN node_modules/webpack-filter-warnings-plugin
npm WARN   webpack-filter-warnings-plugin@"^1.2.1" from @storybook/builder-webpack4@6.4.20
npm WARN   node_modules/@storybook/react/node_modules/@storybook/core/node_modules/@storybook/core-server/node_modules/@storybook/builder-webpack4
npm WARN
npm WARN Conflicting peer dependency: webpack@4.47.0
npm WARN node_modules/webpack
npm WARN   peer webpack@"^2.0.0 || ^3.0.0 || ^4.0.0" from webpack-filter-warnings-plugin@1.2.1
npm WARN   node_modules/webpack-filter-warnings-plugin
npm WARN     webpack-filter-warnings-plugin@"^1.2.1" from @storybook/builder-webpack4@6.4.20
npm WARN     node_modules/@storybook/react/node_modules/@storybook/core/node_modules/@storybook/core-server/node_modules/@storybook/builder-webpack4

> mattermost-desktop@5.7.0-develop.1 postinstall
> node scripts/postinstall.js && electron-builder install-app-deps

  • electron-builder  version=24.12.0
  • loaded configuration  file=/home/andrey/PROJECTS/desktop/electron-builder.json
  • rebuilding native dependencies  dependencies=macos-notification-state@2.0.2, windows-focus-assist@1.3.0 platform=linux arch=x64

up to date, audited 2953 packages in 26s

304 packages are looking for funding
  run `npm fund` for details

62 vulnerabilities (24 moderate, 32 high, 6 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
```

**Comment by Antiarchitect** [comment]
Solved with upgrade to NixOS 24.05. Don't know what exactly did the trick.
